ironic/drivers/modules/drac/management.py

```diff
@@ -183,10 +186,21 @@ def set_boot_device(node, device, persistent=False):
     :raises: DracOperationError on an error from python-dracclient.
     """
 
-    drac_job.validate_job_queue(node)
-
     client = drac_common.get_drac_client(node)
 
+    # If pending BIOS job for changing the boot mode is in job queue,
+    # we need to clear that job before executing clear_job_queue clean step.
+    # Otherwise, any pending config jobs can cause creating new config jobs
+    # to fail.
+    unfinished_jobs = drac_job.list_unfinished_jobs(node)
+    if unfinished_jobs:
+        unfinished_bios_jobs = \
+            [job.id for job in unfinished_jobs if _BIOS_JOB_NAME in job.name]
+        if unfinished_bios_jobs:
+            client.delete_jobs(job_ids=unfinished_bios_jobs)
+        else:
+            drac_job.validate_job_queue(node)
```

Comment:
Just need discuss here one point, this logic will only clear pending bios jobs. In case of other pending jobs, validate_job_queue() will execute and raise an Unfinished config jobs found Error. So other pending jobs won't get deleted from job queue, 
Is this okay ?

Reply:
That's a good point Pradip!  I think the best thing to do is to preserve the prior behavior as much as possible, which was: if there are any unfinished config jobs then raise an exception.
I think we have 3 different cases here:
1. We have no unfinished config jobs
    In this case, we should just continue along so the cleaning step can run
2. We have only an unfinished bios config job
    In this case, I think we should delete it and continue so the cleaning step can run
3. We have more than 1 unfinished config job (this means that we have at least one non-bios config job)
    In this case, we should raise an exception and not delete any jobs
I think that taking this approach will minimally modify the prior behavior, which is probably a good thing.

Reply:
Okay, so as per current logic, case 1 & 2 are working, for case 3 need to change if statement with "if unfinished_bios_jobs and len(unfinished_jobs) == len(unfinished_bios_jobs)"  , am I right?

Reply:
Yes, that will correctly pick out the case where there are only bios bios config jobs, and in that case, we should delete them.  In practice, it should only be possible to have unfinished bios config job at a time, but i think it's a good idea to have the code work for more than 1.
The else clause will then execute for case #3.

Reply:
Okay, will add this change, thanks!

Reply:
Done